packages/flutter/lib/src/widgets/inherited_model.dart

```diff
@@ -0,0 +1,234 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/foundation.dart';
+
+import 'framework.dart';
+
+/// An [InheritedWidget] that's intended to be used as the base class for data
+/// models which are scoped to a widget tree.
+///
+/// An inherited widget's dependents are unconditionally rebuilt when the
+/// inherited widget changes per [InheritedWidget.updateShouldNotify].
```

Comment:
it's not clear here that this is not what this widget does. Maybe `An inherited widget's dependents are normally unconditionally rebuilt when the inherited widget changes per [InheritedWidget.updateShouldNotify]. This widget changes this to only call...` or some such